cdragontoolbox/tftdata.py

```diff
@@ -65,17 +67,70 @@ def replace_in_data(data):
                     if key in data and data[key] in replacements:
                         data[key] = replacements[data[key]]
 
-            for data in template["champs"]:
+            for data in [template["sets"][x]["traits"] for x in template["sets"]]:
+                replace_in_data(data)
+            for data in [template["sets"][x]["champions"] for x in template["sets"]]:
                 replace_in_data(data)
                 if "ability" in data:
                     replace_in_data(data["ability"])
             for data in template["items"]:
                 replace_in_data(data)
-            for data in template["traits"]:
-                replace_in_data(data)
 
             with open(os.path.join(output, f"{lang}.json"), "w", encoding="utf-8") as f:
-                json.dump(template, f, cls=NaiveJsonEncoder, indent=4, sort_keys=True)
+                json.dump(template, f, cls=NaiveJsonEncoder, indent=4)
+
+    def generate_correct_set_info(self, sets, traits, champs):
+
+        set_map = {}
+        for key, value in sets.items():
+            current_set = {"name": value["name"]}
+
+            set_champs = []
+            for char_name in value["internal"]["characters"]:
+                for real_char in champs:
+                    if char_name == real_char["internal"]["name"]:
+                        set_champs.append(real_char)
+
+            set_traits = set([y for x in set_champs for y in x["traits"]])
```

Comment:
No need to convert `set()` arg to a list.
And you don't even need `set()`: `{y for x in set_champs for y in x["traits"]}`